core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -3887,62 +3890,56 @@ protected SqlBuilder getQueryString(boolean isGroupFiltering, boolean isUsername
+                    sqlBuilder.where("U.UM_USER_NAME LIKE LOWER(?)", "%" +
+                            expressionCondition.getAttributeValue());
                 }
-            } else if (ExpressionOperation.SW.toString().equals(expressionCondition.getOperation()) && ExpressionAttribute
-                    .ROLE.toString().equals(expressionCondition.getAttributeName())) {
-                sqlBuilder.where("R.UM_ROLE_NAME LIKE ?", expressionCondition.getAttributeValue() + "%");
-            } else if (ExpressionOperation.SW.toString().equals(expressionCondition.getOperation()) && ExpressionAttribute
-                    .USERNAME.toString().equals(expressionCondition.getAttributeName())) {
+            } else if (ExpressionOperation.SW.toString().equals(expressionCondition.getOperation()) &&
+                    ExpressionAttribute.USERNAME.toString().equals(expressionCondition.getAttributeName())) {
                 if (isCaseSensitiveUsername()) {
                     sqlBuilder.where("U.UM_USER_NAME LIKE ?", expressionCondition.getAttributeValue() + "%");
                 } else {
-                    sqlBuilder.where("U.UM_USER_NAME LIKE LOWER(?)", expressionCondition.getAttributeValue() + "%");
+                    sqlBuilder.where("U.UM_USER_NAME LIKE LOWER(?)", expressionCondition.getAttributeValue()
+                            + "%");
                 }
             } else {
-                if (ExpressionOperation.EQ.toString().equals(expressionCondition.getOperation())) {
-                    sqlBuilder.where("UA.UM_ATTR_NAME = ?", expressionCondition.getAttributeName());
-                    sqlBuilder.where("UA.UM_ATTR_VALUE = ?", expressionCondition.getAttributeValue());
-                } else if (ExpressionOperation.CO.toString().equals(expressionCondition.getOperation())) {
-                    sqlBuilder.where("UA.UM_ATTR_NAME = ?", expressionCondition.getAttributeName());
-                    sqlBuilder.where("UA.UM_ATTR_VALUE LIKE ?", "%" + expressionCondition.getAttributeValue() + "%");
-                } else if (ExpressionOperation.EW.toString().equals(expressionCondition.getOperation())) {
-                    sqlBuilder.where("UA.UM_ATTR_NAME = ?", expressionCondition.getAttributeName());
-                    sqlBuilder.where("UA.UM_ATTR_VALUE LIKE ?", "%" + expressionCondition.getAttributeValue());
-
-                } else if (ExpressionOperation.SW.toString().equals(expressionCondition.getOperation())) {
-                    sqlBuilder.where("UA.UM_ATTR_NAME = ?", expressionCondition.getAttributeName());
-                    sqlBuilder.where("UA.UM_ATTR_VALUE LIKE ?", expressionCondition.getAttributeValue() + "%");
+                // Claim filtering
+                if (!MYSQL.equals(dbType)) {
+                    multiClaimQueryBuilder(sqlBuilder, header, hitClaimFilter, expressionCondition);
+                    hitClaimFilter = true;
+                } else {
+                    //TODO
```

Comment:
Are we tracking this with a Git issue?

Reply:
fixed with later PR, also we track it. #1839